### ISSUE TITLE
Scanning and casting local files from docker container don't work

### ISSUE THREAD

**Issue by steinj** [skorokithakis/catt#300]
I'm setting up a docker container that will cast local files to a chromecast. I have the docker container configured to use host networking. I've used both custom build docker containers as well as the pyhon:3.7 container shown in the README for troubleshooting. There appear to be two issues when interacting with a chromecast: chromecast discoverability and the casting itself.

1. Chromecast Discoverability

`cat scan` doesn't appear to find any local chromecasts. I'm able to interact with a chromecast if I know its IP address though this is less convenient than using the device's name. I.e. the only output I'm able to get reliably is:

```
root@MACHINE:/# catt scan
Scanning Chromecasts...
Error: No devices found
```

2. Casting hangs for local files

When I load media into the docker container, catt outputs the following on the command line though the media doesn't appear on the chromecast.

```
root@MACHINE:/# catt -d CHROMECAST_IP cast MEDIA
Casting local file MEDIA...
Showing "MEDIA" on "NAME OF CHROMECAST"...
Serving local file(s).
```
Instead, the chromecast goes into a mode where it appears to be loading the media from the chromecast since the screen either displays the cast symbol or a blank screen.

----------------------------

I spent some time troubleshooting and saw https://github.com/skorokithakis/catt/issues/274. Exposing ports 45000-47000 didn't appear to help. It's not clear how else I need to configure docker to make the above use cases work properly.

**Comment by skorokithakis** [comment]
I've seen the scanning failure too, unfortunately it seems to be at the Pychromecast level so there isn't much we can do (we use Pychromecast's scanning functionality).

You shouldn't need to do anything other than configure host networking, does casting work when you run catt locally?

**Comment by steinj** [comment]
That's unfortunate that Pychromecast has those issues. After a quick look at their issues backlog it appears to come up a few times.

As for running catt locally, it works flawlessly when I run it directly from the host machine. On the container I'm only able to cast remote files such as youtube. As soon as I use a local file it hangs.

The differences are as follows:

--------------

host machine:
```
user@HOST:~/$ catt -d CHROMECAST_IP cast MEDIA 
Casting local file MEDIA...
Showing "MEDIA" on "NAME_OF_CHROMECAST"...
Serving local file(s).
HOST_IP - - [DATETIME] "GET /?loaded_from_catt HTTP/1.1" 200 - image/gif - 4.52 MB
```
I can see the media playing.

--------------

docker container
```
docker run --network=host -i -t python:3.7 /bin/bash
root@CONTAINER:/# pip install catt
root@CONTAINER:/# catt -d CHROMECAST_IP cast MEDIA
Casting local file MEDIA...
Showing "MEDIA" on "NAME_OF_CHROMECAST"...
Serving local file(s).
```
I copied the media I wanted to play on the chromecast from the host machine for the demonstration. It hangs on the "Serving local file(s)" output.

**Comment by skorokithakis** [comment]
Hmm, that's odd, I'm not sure what might be causing this, using host networking in the container should be indentical to just running it on the host...

**Comment by steinj** [comment]
Agreed. Are you able to reproduce the casting issue? I don't believe there's anything interesting about my network configuration so I'd like to make sure it's reproducible.

**Comment by mcikosos** [comment]
--network=host may be a trick ;-)

**Comment by steinj** [comment]
@mcikosos, I tried that in an earlier comment though it unfortunately didn't work. Were you able to get it working when using `--network=host`?

As an aside, I ended up wrapping calls to `catt` in a proxy that runs outside of the docker container though I must admit this isn't ideal (i.e. it's tech debt).

**Comment by mcikosos** [comment]
@steinj  ohh I'm blind sorry 
In my case scan and casting local files  works in docker using host networking
